functionality.ts

```diff
@@ -0,0 +1,141 @@
+class Calculator {
+  previousOperandTextElement: HTMLElement;
+  currentOperandTextElement: HTMLElement;
+  currentOperand!: string;
+  previousOperand!: string;
+  operation!: string;
+
+  constructor(previousOperandTextElement, currentOperandTextElement) {
+    this.previousOperandTextElement = previousOperandTextElement;
+    this.currentOperandTextElement = currentOperandTextElement;
+    this.clear();
+  }
+
+  clear() {
+    this.currentOperand = "";
+    this.previousOperand = "";
+    this.operation = undefined;
+  }
+
+  delete() {
+    this.currentOperand = this.currentOperand.toString().slice(0, -1);
+  }
+
+  appendNumber(number) {
+    if (number === "." && this.currentOperand.toString().includes(".")) return;
+    this.currentOperand = this.currentOperand.toString() + number.toString();
+  }
+
+  chooseOperation(operation) {
+    if (this.currentOperand === "") return;
+    if (this.previousOperand !== "") {
+      this.compute();
+    }
+    this.operation = operation;
+    this.previousOperand = this.currentOperand;
+    this.currentOperand = "";
+  }
+  compute() {
+    let computation;
```

Comment:
It's best to initialize variables with a value, even if it's a falsy one (0, "", false, undefined, null)